enhance application logs (#2946)
elastic/observability-docs: docs/en/observability/application-logs.asciidoc

[[application-logs]]
= Application logs

Application logs provide valuable insight into events that have occurred within your services and applications.

> The ingestion of those logs requires to deal with two aspects:
> 
> - <<log-types,log format>>: plaintext or ECS (structured JSON) logs.
> - <<ingest-application-logs,logs ingestion>>: how those logs are sent and written to Elasticsearch using a tool like {filebeat}.

Reviewer: Instead of ingestion requiring format and ingestion, what about monitoring requiring format and ingestion?

```suggestion
There are two aspects to consider when monitoring application logs:

- <<log-types,log format>>: are your logs plaintext or Elastic Common Schema (ECS) logs (structured JSON)
- <<ingest-application-logs,log ingestion>>: the tool you use to send and write logs to Elasticsearch
```

> The choice of the log format influences the log ingestion strategy.
> For simplicity, this documentation uses {filebeat} for reference, but any log ingestion tool could be used with equivalent configuration.

Reviewer: ```suggestion
The format of your logs (structured or plaintext) influences your log ingestion strategy.
For simplicity, this documentation uses {filebeat} for ingestion, but any log ingestion tool can be used with an equivalent configuration.
```

Application logs can be correlated to APM services with <<log-correlation,log correlation>>.

Using an APM agent provides extra features:

> - <<log-correlation,log correlation>>: ability to correlate logs to APM traces

Reviewer: ```suggestion
- <<log-correlation,log correlation>>: correlate logs to APM traces
```

{"@timestamp":"2019-08-06T14:08:40.199Z", "log.level":"DEBUG", "message":"init find form", "service.name":"spring-petclinic","process.thread.name":"http-nio-8080-exec-8","log.logger":"org.springframework.samples.petclinic.owner.OwnerController","transaction.id":"28b7fb8d5aba51f1","trace.id":"2869b25b5469590610fea49ac04af7da"}
----

> There are two ways to make the application write ECS formatted logs
> - using {ecs-logging-ref}/intro.html[ECS logging] library
> - using an APM agent and <<log-reformatting,log reformatting>>

Reviewer: ```suggestion
There are two ways to make your application write ECS-formatted logs:
- use an {ecs-logging-ref}/intro.html[ECS logging] library
- use an APM agent and <<log-reformatting,log reformatting>>
```

There are four main ways to ingest application logs into the {stack}.
Your use case will help determine the method that's right for you.

> Here {filebeat} is used for reference, but any ingestion tool
> could be used with equivalent configuration.

Reviewer: ```suggestion
Here, {filebeat} is used for reference, but any ingestion tool
can be used with an equivalent configuration.
```
